site/docs/v1/tech/tutorials/integrate-ruby-rails-api.md

```diff
@@ -0,0 +1,193 @@
+
+```shell
+bundle install
+```
+
+Next, create a file called `.env.development` and insert the following into it.
+
+```ini
+{% remote_include https://raw.githubusercontent.com/FusionAuth/fusionauth-example-rails-api-guide/main/.env.development %}
+```
+
+You can now start writing the code for your Rails API. First, let's create a controller which gives back a JSON message. Create a new file called `app/controllers/messages_controller.rb`, then add the following code:
+
+```ruby
+{% remote_include https://raw.githubusercontent.com/FusionAuth/fusionauth-example-rails-api-guide/main/app/controllers/messages_controller.rb %}
+```
+
+This controller returns a JSON array with messages.
+
+Next, update the `config/routes.rb` file to look like this:
+
+```ruby
+{% remote_include https://raw.githubusercontent.com/FusionAuth/fusionauth-example-rails-api-guide/main/config/routes.rb %}
+```
+
+This tells {{page.technology}} to return the content generated by the `messages_controller.rb` file when the `/messages` path is set up.
+
+You can now start up your server. You should do it in a new terminal window so that you can continue to edit the {{page.technology}} code.
+
+```shell
+bundle e rails s -p 4001
+```
+
+And visit [http://localhost:4001/messages](http://localhost:4001/messages) and view the JSON.
+
+Next, let's configure the token protection for this API. Go back to your previous terminal window and create a `config/initializers/jwt_rack.rb` file, and update it to look like this:
+
+```ruby
+{% remote_include https://raw.githubusercontent.com/FusionAuth/fusionauth-example-rails-api-guide/main/config/initializers/jwt_rack.rb %}
+```
+
+This tells {{page.technology}} to check for various attributes of the token, including the `iss` and the `aud`. Read the [`rack_jwt` gem documentation](https://github.com/FusionAuth/rack-jwt/) for more.
+
+You can also access the JWT in the controller. Below, the messages controller adds a special message if the user has a certain role.
+
+```ruby
+{% remote_include https://raw.githubusercontent.com/FusionAuth/fusionauth-example-rails-api-guide/main/app/controllers/messages_controller.rb %}
+```
+
+Now, back to the terminal where your server is running. Stop it (using `control-C`) and restart it.
+
+```shell
+bundle e rails s -p 4001
+```
+
+Visit [http://localhost:4001/messages](http://localhost:4001/messages), you'll get an error:
+
+```json
+{"error":"Invalid JWT token : Decode Error"}
+```
+
+Your API is protected. Now, let's get an access token so authorized clients can get the API results.
+
+## Testing the API Flow
+
+There are a number of ways to get an access token, as mentioned, but for clarity, let's use the login API to mimic a client. 
+
+Run this command in a terminal:
+
+```shell
+curl -H 'Authorization: YOUR_API_KEY_FROM_ABOVE' \
+     -H 'Content-type: application/json' \
+     -d '{"loginId": "admin@example.com", "password":"password","applicationId": "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e"} \
```

Comment:
We should call out that this should be an existing user. `admin@example.com` won't exist here and likely the only user that has been set up is the admin user created at the start.

also 
```suggestion
     -d '{"loginId": "admin@example.com", "password":"password","applicationId": "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e"}' \
```

Reply:
Great catch.